test/app-tap/module_api.test.lua

```diff
@@ -252,7 +252,12 @@ require('tap').test("module_api", function(test)
     test:test("buffers", test_buffers, module)
     test:test("tuple_validate", test_tuple_validate, module)
 
+    test:ok(module.session_check_admin(), "session_check_admin is ok")
+    test:ok(module.session_check_fiber(), "session_check_fiber is ok")
+
     space:drop()
+
+    test:ok(module.session_check_after_cleanup(), "session_check_after_cleanup is ok")
```

Comment:
- You didn't test session peer when there is actually a peer;
- You didn't test getting peer of a not existing session;
- You didn't test having a not big enough buffer for a peer name;
- You didn't test getting UID when the user is deleted (you can drop the user after doing `su` to him, AFAIR);
- You didn't test getting EUID, and that it might be different from UID;
- You didn't test getting session type.

Reply:
As I mentioned in previous comments the idea was to reproduce tests from `app-tap/module_api.test.lua`. Not all of them can be reproduced, because no everything from `box.session` Lua API is added to Module API in this PR (for example, `box.session.su` is not added). All 3 tests, that you see currently are snippets from `app-tap/module_api.test.lua`.

What is you opinion on this? What tests from list above must be added?

Reply:
Personally, I would not look at module_api.test.lua as an example. I would try to write new tests, easier to read and extend. But covering at least the same cases and maybe more. Although it does not mean you need to do it (repeat something != justification, including blindly doing what I said). Maybe you can copy logic from the module_api.test.lua test but do some minor changes to address the strange things I noticed and questions I asked in the other comments.

As for the mentioned list of tests in the comment above - I think you need to cover them all. They do not look hard to test, rather easy, but otherwise you simply have code not tested a single time anywhere. We can't merge such code (unless it is urgent and some managers are running for their yearly bonus of course, haha).